qiskit/circuit/quantumcircuit.py

```diff
@@ -2205,27 +2205,41 @@ def measure_active(self, inplace: bool = True) -> Optional["QuantumCircuit"]:
         else:
             return None
 
-    def measure_all(self, inplace: bool = True) -> Optional["QuantumCircuit"]:
+    def measure_all(
+        self, inplace: bool = True, add_creg: bool = True
+    ) -> Optional["QuantumCircuit"]:
         """Adds measurement to all qubits. Creates a new ClassicalRegister with a
-        size equal to the number of qubits being measured.
+        size equal to the number of qubits being measured if `add_creg = True`.
 
         Returns a new circuit with measurements if `inplace=False`.
 
         Args:
             inplace (bool): All measurements inplace or return new circuit.
+            add_creg (bool): Adds a new classical register.
 
         Returns:
             QuantumCircuit: Returns circuit with measurements when `inplace = False`.
+
+        Raises:
+            CircuitError: if `add_creg = False` and `len(circ.clbits) < len(circ.qubits)`.
```

Comment:
If possible, it's better to describe the condition in words rather than code, because it's easier for people to understand, usually.  So perhaps we could say something like

```python
Raises:
    CircuitError: if ``add_bits=False`` but there are not enough classical bits in the circuit.
```